src/processor/map/map_semi_masker.cpp

```diff
@@ -44,9 +44,11 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapSemiMasker(LogicalOperator* log
             KU_ASSERT(physicalOp->getChild(0)->getOperatorType() == PhysicalOperatorType::GDS_CALL);
             auto gds = physicalOp->getChild(0)->ptrCast<GDSCall>();
             KU_ASSERT(gds->hasSemiMask());
-            auto mask = gds->getSemiMask();
-            auto tableID = mask->getNodeTable()->getTableID();
-            masksPerTable.at(tableID).emplace_back(mask, 0);
+            // TODO(Xiyang): we can refactor this logic with RECURSIVE JOIN above.
```

Comment:
Do you want to address this TODO while you're at it? You might leave algorithms anyways.

Reply:
I'll do this in a separate PR.